### ISSUE TITLE
2021.8 version doesn't follow transition length setting?

### ISSUE THREAD

**Issue by Protoncek** [JeffResc/Sonoff-D1-Dimmer#8]
Hi!
Yesterday i've had to do some changes in my dimmer, so i had to replace file with your's latest (i use "d1_dimmer_no_rf"). to avoid errors. From now on it seems that "transition_length" is not taken into account ? I have it set to 0.1s on one button (so that light is ON immediately) which worked before 2021.8, now transition is fixed appr. 2s. Turn off is also not momentary, but it's somewhat faster than turn on, say half of "ON time", around 1s.
Is that a problem in your new file or in ESPHome?

**Comment by JeffResc** [comment]
Hey, thanks for your report!

As you can see [here](https://github.com/JeffResc/Sonoff-D1-Dimmer/commit/0f5fb859b7ee2cd256d84eb14846234d66bb5834), the only thing that changed with my update is the way that the light traits are set in the internal ESPHome API. I'm not too sure they changed something with their update, but to their knowledge, this is not something I changed in my code or intended to do. I'm going to consult with a few ESPHome experts and see if we can find anything. I'll post updates here as they become available.

**Comment by Protoncek** [comment]
Yeah, well, if you find anything related it would be great, but otherwise don't bother too much, since 2s time is kinda most optimal anyway (at least for me). Now i've got used to 2 sec, so it's quite fine.  
It's interesting, though, why is this... and, am i the only one with this problem (since no one else posted same problem yet).
